src/router.js

```diff
@@ -148,28 +193,19 @@ class Router {
    */
   updateRoute(route) {
     return new Promise((resolve, reject) => {
-      if (!this.findRouteById(route._id)) {
-        return this
+      if (!route._id) {
+        return resolve(this
```

Comment:
écrit plutot return resolve(this.addRoute()); car tu peux renvoyer un promise, c'est géré